### PR TITLE
домашне завдання №5.1

### PR DESCRIPTION
**Pull request by OlhaVal**
Hw 5.1

### CODE REVIEW

File: lesson_05/homework_05_1.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  'Nissan Titan': ('silver', 2018, 5.6, 'pickup', 35000)
+}
+search_criteria = (2017, 1.6, 36000)
+
+new_car_data = []
+dict_car_data = {}
+
+for key, value in car_data.items():
+    if (value[1] >= search_criteria[0]) and (value[2] >= search_criteria[1]) and (value[-1] <= search_criteria[2]):
+        dict_car_data.update({key: value[-1]})
```
Comment (DartBrovsky) [inline lesson_05/homework_05_1.py:54]: Hi, it is not good practice to update dict values while iteration by dict. It is better to use dict comprehension with if statement for search criteria. For ex:
new_dict = {key: value for key, value in old_dict if complex_statement}

**Review by DartBrovsky — CHANGES_REQUESTED** [review]
Please take a look to my comment regarding dict comprehension. In general - your solution is okay)

Thanks.

**Review by OlhaVal — COMMENTED** [review]
It was an interesting search for code optimization :)

File: lesson_05/homework_05_1.py (hunk 1 above)
Comment (OlhaVal) [reply lesson_05/homework_05_1.py]: @DartBrovsky, Thanks for the advice.  I was really some chelenge to understend this option. 
Today's lecture was very helpful

**Review by DartBrovsky — APPROVED** [review]

**Review by DartBrovsky — APPROVED** [review]
We are good. Your solution is okay. I approve it, thanks.

**Pull request merged**